httputil/client.go

```diff
@@ -33,6 +41,55 @@ type ClientOptions struct {
 	Proxy      func(*http.Request) (*url.URL, error)
 }
 
+func addLocalSslCertificates(conf *tls.Config) (allCAs *x509.CertPool, err error) {
+	if conf != nil && conf.RootCAs != nil {
+		allCAs = conf.RootCAs
+	} else {
+		allCAs, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("cannot read system certificates: %v", err)
+		}
+	}
+	if allCAs == nil {
+		return nil, fmt.Errorf("cannot use empty certificates pool")
+	}
+	extraCertFiles, err := filepath.Glob(filepath.Join(dirs.SnapdExtraSslCertsDir, "*.pem"))
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range extraCertFiles {
+		extraCert, err := ioutil.ReadFile(p)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read extra certificate: %v", err)
+		}
+		// XXX: continue here trying to add others?
+		if ok := allCAs.AppendCertsFromPEM(extraCert); !ok {
+			return nil, fmt.Errorf("cannot append extra ssl certificate: %v", p)
+		}
+	}
+	return allCAs, nil
+}
+
+type dialTLS struct {
```

Comment:
For the sake of documenting this, is the point of this type to provide custom certificates and having a `dial` method? If so can you spare a comment please?
```suggestion
// dialTLS implements <iface> while honoring snapd-only TLS certificates.
type dialTLS struct {
```

Reply:
Added some docstrings/comments, hope this is clearer with them.